src/main/java/org/apache/sling/dynamicinclude/api/IncludeGenerator.java

```diff
@@ -17,13 +17,15 @@
  * under the License.
  */
 
-package org.apache.sling.dynamicinclude.generator;
+package org.apache.sling.dynamicinclude.api;
+
+import org.apache.sling.api.SlingHttpServletRequest;
 
 /**
  * Include generator interface
  */
 public interface IncludeGenerator {
     String getType();
 
-    String getInclude(String url);
+    String getInclude(SlingHttpServletRequest request,String url);
```

Comment:
I think it's confusing to pass in the request and a string named `url` I see that the `url` parameter is based on the sling request, so maybe we can have a different approach here.

Reply:
the issue here is that the url is not just taken from the request as it is. It is previously checked to do things like removing the query part, map it,encode, etc (in https://github.com/apache/sling-org-apache-sling-dynamic-include/blob/master/src/main/java/org/apache/sling/dynamicinclude/IncludeTagFilter.java#L149),.
we could remove the url parameter, but that means any IncludeGenerator would be responsible for doing that

Reply:
I think that would be ok, granted that we:

- find a better name for it ( normalizedUrl?)
- add better documentation to the interface and the method, explaining why we pass in an apparently redundant parameter